tests/tests_ssh.yml

```diff
@@ -61,5 +61,19 @@
       delegate_to: "{{ kdump_test_ssh_server_outside }}"
       delegate_facts: true
 
+    - name: >-
+        Fix /sbin/mkdumprd to point netdev to eth0.
+        netdev is the device that mkdumprd must use to connect to the remote
+        server to store kdump logs.
+        On EL 6, mkdumprd does not find netdev correctly when connecting via
+        ssh to localhost.
+      replace:
+        path: /sbin/mkdumprd
+        regexp: netdev=`\/sbin\/ip route get to \$remoteip 2>&1`
+        replace: netdev=eth0
+      when:
+        - ansible_distribution in ['CentOS','RedHat']
+        - ansible_distribution_major_version == "6"
```

Comment:
I don't think that a test is expected to change its system in this way (modifying a system executable that is not a configuration file). Imagine a test running together with many other tests.
I believe that if you can't make the test work properly in some conditions, it is better to skip it.

Reply:
That's right, while eth0 is the device to use in our CI, it might be anything else in some other environment. I must somehow skip the tests_ssh.yml test on EL 6. I do not want to remove the support for EL 7 completely though, as it works just fine until you make kdump ssh it itself. I need to find a way to skip one test playbook on EL 7.

Reply:
What's the problem with EL 7?

Reply:
Sorry, I meant EL 6 the whole time. The problem is, it's mkdumprd does not expect sshing to localhost. And when we do configure kdump to store logs via ssh to a localhost, it fails finding the correct interface.

Reply:
Yes, I understand the problem.
To skip the test in certain condition, you can use `meta: end_host`. Some examples are here: https://github.com/linux-system-roles/metrics/pull/53 .
I recommend to skip the test on EL 6 only if `kdump_test_ssh_server_outside` is not equal to `inventory_hostname`, because otherwise the test should work on EL 6 just fine. (More precisely, the condition should be that `kdump_test_ssh_server_outside` should not be a local IP address for the managed host, but I am not sure how to express this condition.)